File: frontend/src/app/model/Product.ts
export class Product {

  private _id: number;
  private _productName: string;
  private _description: string;
  private _price: number;
  private _model: string;
  private _brand: string;
  private _ageOfRelease: string;
  private _countInStock: number;
  private _productCategory: string;
  private _productStatus: string;
  private _previewImage: File;

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get previewImage(): File {
    return this._previewImage;
  }

  set previewImage(value: File) {
    this._previewImage = value;
  }

  get productName(): string {
    return this._productName;
  }

  set productName(value: string) {
    this._productName = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get price(): number {
    return this._price;
  }

  set price(value: number) {
    this._price = value;
  }

  get model(): string {
    return this._model;
  }

  set model(value: string) {
    this._model = value;
  }

  get brand(): string {
    return this._brand;
  }

  set brand(value: string) {
    this._brand = value;
  }

  get ageOfRelease(): string {
    return this._ageOfRelease;
  }

  set ageOfRelease(value: string) {
    this._ageOfRelease = value;
  }

  get countInStock(): number {
    return this._countInStock;
  }

  set countInStock(value: number) {
    this._countInStock = value;
  }

  get productCategory(): string {
    return this._productCategory;
  }

  set productCategory(value: string) {
    this._productCategory = value;
  }

  get productStatus(): string {
    return this._productStatus;
  }

  set productStatus(value: string) {
    this._productStatus = value;
  }
}

File: backend/src/main/java/by/kapitonov/computer/shop/backend/model/product/Product.java
package by.kapitonov.computer.shop.backend.model.product;

import by.kapitonov.computer.shop.backend.model.AuditEntity;
import by.kapitonov.computer.shop.backend.model.Image;
import by.kapitonov.computer.shop.backend.model.PaymentMethod;
import by.kapitonov.computer.shop.backend.model.product.detail.ProductCategory;
import by.kapitonov.computer.shop.backend.model.product.detail.ProductStatus;
import by.kapitonov.computer.shop.backend.model.Review;
import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.*;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;


@Table(name = "products")
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
@ToString
@Builder
@EntityListeners(AuditingEntityListener.class)
public class Product extends AuditEntity {

    @Column(name = "product_name", nullable = false, length = 50)
    private String productName;

    @Column(name = "description", nullable = false, length = 255)
    private String description;

    @Column(name = "price", nullable = false, length = 100)
    private Integer price;

    @Column(name = "model", nullable = false, length = 100)
    private String model;

    @Column(name = "brand", nullable = false, length = 100)
    private String brand;

    @Column(name = "age_of_release", nullable = false, length = 10)
    private String ageOfRelease;

    @Column(name = "count_in_stock", nullable = false, length = 255)
    private Integer countInStock;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "preview_image_id", referencedColumnName = "id")
    private Image previewImage;

    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JoinTable(name = "product_images",
            joinColumns = {@JoinColumn(name = "product_id", referencedColumnName = "id")},
            inverseJoinColumns = {@JoinColumn(name = "image_id", referencedColumnName = "id")}
    )
    private List<Image> images;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_category_id", referencedColumnName = "id")
    private ProductCategory productCategory;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "status_id", referencedColumnName = "id")
    private ProductStatus productStatus;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonBackReference
    private List<Review> reviews = new ArrayList<>();

}
